web/src/rest/publish.js

```diff
@@ -56,6 +56,20 @@ const publishRest = new Vue({
       await oauthClient.logout();
       this.user = null;
     },
+    async apiKey() {
+      try {
+        const { data } = await client.get('auth/token/');
+        return data;
+      } catch (e) {
+        // If the request returned 404, the user doesn't have an API key yet
+        if (e.message === 'Request failed with status code 404') {
```

Comment:
Wouldn't it be a bit better to directly check the response status? I believe it'd be something like:

```suggestion
        if (e.response.status === 404) {
```

However I'm not certain.